File: shared/src/iosMain/kotlin/com/d1onix/dishlab/Previews.kt
package com.d1onix.dishlab

import com.d1onix.dishlab.designsystem.preview.MiseComponentGalleryController
import com.d1onix.dishlab.feature.home.presentation.homePreviewController
import com.d1onix.dishlab.feature.products.presentation.graphPreviewController
import com.d1onix.dishlab.feature.products.presentation.graphSheetPreviewController
import com.d1onix.dishlab.feature.products.presentation.historyPreviewController
import com.d1onix.dishlab.feature.recipes.presentation.cookingPreviewController
import com.d1onix.dishlab.feature.recipes.presentation.recipeDetailPreviewController
import com.d1onix.dishlab.feature.recipes.presentation.recipeListPreviewController
import com.d1onix.dishlab.feature.scanner.presentation.scanManualEntryPreviewController
import com.d1onix.dishlab.feature.scanner.presentation.scanNotFoundPreviewController
import com.d1onix.dishlab.feature.scanner.presentation.scanPreviewController
import platform.UIKit.UIViewController

/**
 * The iOS preview catalogue, as seen from Swift.
 *
 * Feature modules are `implementation` dependencies of `:shared`, so their own
 * preview controllers are not part of the `Shared` framework. Listing them here
 * is what exports them, and it keeps the framework's public surface a
 * deliberate choice rather than everything every feature happens to declare.
 *
 * `iosApp/iosApp/Previews.swift` turns each of these into a SwiftUI `#Preview`.
 * None of them touch the DI graph: they render a fixed state, so opening the
 * canvas never starts a repository, a data store or the camera.
 */
object DishLabPreviews {

    fun home(): UIViewController = homePreviewController()

    fun scan(): UIViewController = scanPreviewController()

    fun scanManualEntry(): UIViewController = scanManualEntryPreviewController()

    fun scanNotFound(): UIViewController = scanNotFoundPreviewController()

    fun graph(): UIViewController = graphPreviewController()

    fun graphSheet(): UIViewController = graphSheetPreviewController()

    fun history(): UIViewController = historyPreviewController()

    fun recipeList(): UIViewController = recipeListPreviewController()

    fun recipeDetail(): UIViewController = recipeDetailPreviewController()

    fun cooking(): UIViewController = cookingPreviewController()

    fun componentGallery(): UIViewController = MiseComponentGalleryController()
}

File: iosApp/iosApp/Previews.swift
import SwiftUI
import Shared

/// Hosts one Compose screen from `DishLabPreviews` in the SwiftUI canvas.
///
/// There is no Compose preview renderer on iOS, so an iOS preview is the real
/// framework running in the simulator — which is the point: this is where the
/// safe-area insets, the keyboard, and the actual font rasterisation show up.
/// Android's `@Preview` functions live in each feature's `androidMain` and
/// cover the per-state matrix instead; these cover the platform.
///
/// The controllers render fixed state and never build the DI graph, so opening
/// the canvas starts no repository, no data store and no camera.
private struct ComposePreview: UIViewControllerRepresentable {
    let controller: () -> UIViewController

    func makeUIViewController(context: Self.Context) -> UIViewController {
        controller()
    }

    func updateUIViewController(_ uiViewController: UIViewController, context: Self.Context) {}
}

#Preview("Home") {
    ComposePreview { DishLabPreviews.shared.home() }.ignoresSafeArea()
}

#Preview("Scan") {
    ComposePreview { DishLabPreviews.shared.scan() }.ignoresSafeArea()
}

#Preview("Scan · manual entry") {
    ComposePreview { DishLabPreviews.shared.scanManualEntry() }.ignoresSafeArea()
}

#Preview("Scan · not found") {
    ComposePreview { DishLabPreviews.shared.scanNotFound() }.ignoresSafeArea()
}

#Preview("Graph") {
    ComposePreview { DishLabPreviews.shared.graph() }.ignoresSafeArea()
}

#Preview("Graph · detail sheet") {
    ComposePreview { DishLabPreviews.shared.graphSheet() }.ignoresSafeArea()
}

#Preview("History") {
    ComposePreview { DishLabPreviews.shared.history() }.ignoresSafeArea()
}

#Preview("Recipes") {
    ComposePreview { DishLabPreviews.shared.recipeList() }.ignoresSafeArea()
}

#Preview("Recipe detail") {
    ComposePreview { DishLabPreviews.shared.recipeDetail() }.ignoresSafeArea()
}

#Preview("Cooking") {
    ComposePreview { DishLabPreviews.shared.cooking() }.ignoresSafeArea()
}

#Preview("Component gallery") {
    ComposePreview { DishLabPreviews.shared.componentGallery() }.ignoresSafeArea()
}
